File: internal/models/building.go
package models

import "time"

// BuildingSnapshot represents a snapshot of a player's building
type BuildingSnapshot struct {
	PlayerTag    string    `json:"player_tag"`
	BuildingType string    `json:"building_type"`
	Level        int       `json:"level"`
	SnapshotTime time.Time `json:"snapshot_time"`
}

// BuildingUpgrade represents a building upgrade event
type BuildingUpgrade struct {
	PlayerTag    string    `json:"player_tag"`
	PlayerName   string    `json:"player_name"`
	BuildingType string    `json:"building_type"`
	OldLevel     int       `json:"old_level"`
	NewLevel     int       `json:"new_level"`
	UpgradeTime  time.Time `json:"upgrade_time"`
}

// BuildingTracker represents tracking information for building monitoring
type BuildingTracker struct {
	TelegramID           int64     `json:"telegram_id"`
	PlayerTag            string    `json:"player_tag"`
	LastCheck            time.Time `json:"last_check"`
	NotificationsEnabled bool      `json:"notifications_enabled"`
}

// NewBuildingSnapshot creates a new BuildingSnapshot instance
func NewBuildingSnapshot(playerTag, buildingType string, level int) *BuildingSnapshot {
	return &BuildingSnapshot{
		PlayerTag:    playerTag,
		BuildingType: buildingType,
		Level:        level,
		SnapshotTime: time.Now(),
	}
}

// NewBuildingUpgrade creates a new BuildingUpgrade instance
func NewBuildingUpgrade(playerTag, playerName, buildingType string, oldLevel, newLevel int) *BuildingUpgrade {
	return &BuildingUpgrade{
		PlayerTag:    playerTag,
		PlayerName:   playerName,
		BuildingType: buildingType,
		OldLevel:     oldLevel,
		NewLevel:     newLevel,
		UpgradeTime:  time.Now(),
	}
}

// NewBuildingTracker creates a new BuildingTracker instance
func NewBuildingTracker(telegramID int64, playerTag string) *BuildingTracker {
	return &BuildingTracker{
		TelegramID:           telegramID,
		PlayerTag:            playerTag,
		LastCheck:            time.Now(),
		NotificationsEnabled: true,
	}
}

File: src/models/building.py
"""
Модели данных для зданий и отслеживания улучшений
"""
from dataclasses import dataclass
from typing import List, Dict, Optional
from datetime import datetime


@dataclass
class BuildingSnapshot:
    """Снимок состояния зданий игрока"""
    player_tag: str
    snapshot_time: str
    buildings_data: str  # JSON строка с данными о зданиях
    
    def __init__(self, player_tag: str, snapshot_time: str, buildings_data: str):
        self.player_tag = player_tag
        self.snapshot_time = snapshot_time
        self.buildings_data = buildings_data


@dataclass
class BuildingUpgrade:
    """Информация об улучшении здания"""
    building_name: str
    old_level: int
    new_level: int
    
    def __init__(self, building_name: str, old_level: int, new_level: int):
        self.building_name = building_name
        self.old_level = old_level
        self.new_level = new_level


@dataclass
class BuildingTracker:
    """Настройки отслеживания улучшений для пользователя"""
    telegram_id: int
    player_tag: str
    is_active: bool
    created_at: str
    last_check: Optional[str] = None
    
    def __init__(self, telegram_id: int, player_tag: str, is_active: bool, 
                 created_at: str, last_check: Optional[str] = None):
        self.telegram_id = telegram_id
        self.player_tag = player_tag
        self.is_active = is_active
        self.created_at = created_at
        self.last_check = last_check